File: micronaut/src/main/java/io/swagger/model/V3StopGps.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.validation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * V3StopGps
 */
@Validated
@Introspected
@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.MicronautCodegen", date = "2024-04-14T03:36:01.894111500+10:00[Australia/Sydney]")
public class V3StopGps   {
  @JsonProperty("latitude")
  private Float latitude = null;

  @JsonProperty("longitude")
  private Float longitude = null;

  public V3StopGps latitude(Float latitude) {
    this.latitude = latitude;
    return this;
  }

  /**
   * Geographic coordinate of latitude at stop
   * @return latitude
  **/
  @Schema(description = "Geographic coordinate of latitude at stop")

  public Float getLatitude() {
    return latitude;
  }

  public void setLatitude(Float latitude) {
    this.latitude = latitude;
  }

  public V3StopGps longitude(Float longitude) {
    this.longitude = longitude;
    return this;
  }

  /**
   * Geographic coordinate of longitude at stop
   * @return longitude
  **/
  @Schema(description = "Geographic coordinate of longitude at stop")

  public Float getLongitude() {
    return longitude;
  }

  public void setLongitude(Float longitude) {
    this.longitude = longitude;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3StopGps v3StopGps = (V3StopGps) o;
    return Objects.equals(this.latitude, v3StopGps.latitude) &&
        Objects.equals(this.longitude, v3StopGps.longitude);
  }

  @Override
  public int hashCode() {
    return Objects.hash(latitude, longitude);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3StopGps {\n");
    
    sb.append("    latitude: ").append(toIndentedString(latitude)).append("\n");
    sb.append("    longitude: ").append(toIndentedString(longitude)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: dart/lib/model/v3_stop_gps.dart
part of swagger.api;

class V3StopGps {
  /* Geographic coordinate of latitude at stop */
  double latitude = null;
/* Geographic coordinate of longitude at stop */
  double longitude = null;

  V3StopGps();

  @override
  String toString() {
    return 'V3StopGps[latitude=$latitude, longitude=$longitude, ]';
  }

  V3StopGps.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    latitude = json['latitude'];
    longitude = json['longitude'];
  }

  Map<String, dynamic> toJson() {
    return {
      'latitude': latitude,
      'longitude': longitude
     };
  }

  static List<V3StopGps> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3StopGps>() : json.map((value) => new V3StopGps.fromJson(value)).toList();
  }

  static Map<String, V3StopGps> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3StopGps>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3StopGps.fromJson(value));
    }
    return map;
  }
}
